ProcessLib/ProcessOutput.cpp

```diff
@@ -214,13 +199,14 @@ void doProcessOutput(
         }
     };
 
-    for (auto const& elem : secondary_variables)
+    for (auto const& external_variable_name : output_variables)
     {
-        auto const& var_name = elem.first;
-        if (output_variables.find(var_name) != output_variables.cend())
-        {
-            add_secondary_var(elem.second, var_name);
-        }
+        if (already_output.find(external_variable_name) != already_output.end())
+            continue;
+        already_output.insert(external_variable_name);
```

Comment:
Just technical thing:
Instead of searching and inserting, you could use the return value of std::map::insert, which is a pair of iterator and boolean, the latter indicating success/failure of the insertion.